model/project_aliases.go

```diff
@@ -49,11 +50,16 @@ const (
 // all tasks containing the string “test” on all variants containing the string
 // “linux”; and to run all tasks beginning with the string “compile” to run on all
 // variants beginning with the string “ubuntu1604”.
+
+// Git tags use a special alias "__git_tag" and create a new version for the matching
+// variants/tasks, assuming the tag matches the defined git_tag regex.
+// In this way, users can define different behavior for different kind of tags.
 type ProjectAlias struct {
 	ID          mgobson.ObjectId `bson:"_id" json:"_id"`
 	ProjectID   string           `bson:"project_id" json:"project_id"`
 	Alias       string           `bson:"alias" json:"alias"`
 	Variant     string           `bson:"variant,omitempty" json:"variant"`
+	GitTag      string           `bson:"git_tag" json:"git_tag"`
```

Comment:
Could you explain what you plan on using this for? Is it that we want to run a different set of tasks when you push a tag vs when you push a commit?

Reply:
I can change the comment above this if it wasn't clear: the  purpose is so you can run a different set of tasks for different kind of git tags (so if I have release-candidate-v1.0 versus do-release-v1.0, those could have different tasks run).

Reply:
No need probably, just wanted to know what you intended for this.

Reply:
Would you also be able to specify different YAMLs for different tag sets?

Reply:
Ah interesting point. I don't see why not.

Reply:
Although in that case if multiple tag sets match I think it might be safest to error 🤔

Reply:
Ah, I meant that for tag set A you use YAML A and for tag set B to use YAML B. In the case of multiple matches I think it's fine to error.